docs: what's new, add traces and k6 integration (#75401)
grafana/grafana: docs/sources/whatsnew/whats-new-next/index.md

Welcome to Grafana Cloud! Read on to learn about the newest changes to Grafana Cloud.

## Distributed Tracing in Grafana Cloud k6

<!-- Heitor Tashiro Sergent -->

> <!-- OSS, Enterprise -->

Reviewer: Is this just something left over or some type of tag?  The feature is only available in Grafana cloud - specifically for customers visualizing results in GCk6 and using GCT

Author: Copy+paste that I didn't pay attention to 😓 , thanks for catching that. 🙏

_Generally available in Grafana Cloud_

You can now use the Grafana Cloud Traces integration with Grafana Cloud k6 to quickly debug failed performance tests and proactively improve application reliability.

> Distributed tracing in Grafana Cloud k6 works by having k6 automatically inject tracing metadata into the requests it sends to users’ backend services when they run a test. The tracing data is then correlated with k6 test run data, so users can understand how their services and operations behaved during the whole test run. The collected tracing data is aggregated to generate real-time metrics, such as frequency of calls, error rates, and percentile latencies, that help users narrow their search space and quickly spot anomalies.

Reviewer: >by having k6 automatically inject tracing metadata into the requests

It's automatic if you configure the option.  It also requires that the customer is using Grafana Cloud Traces.  I think we should reword something here for this clarity

Welcome to Grafana Cloud! Read on to learn about the newest changes to Grafana Cloud.

> ## Distributed Tracing in Grafana Cloud k6

Reviewer 2: ```suggestion
## Distributed tracing in Grafana Cloud k6
```

<!-- Heitor Tashiro Sergent -->

_Generally available in Grafana Cloud_

You can now use the Grafana Cloud Traces integration with Grafana Cloud k6 to quickly debug failed performance tests and proactively improve application reliability.

Distributed tracing in Grafana Cloud k6 only requires two things:

- An application instrumented for tracing with Grafana Cloud Traces.

> - Adding a few lines of code to your existing k6 scripts.

Reviewer 2: This is a nit, but for parallel structure, I'd want this to be the noun form. But as I said, a nit and it makes it longer, so take it or leave it.
```suggestion
- The addition of a few lines of code to your existing k6 scripts.
```

Author: I'm going to leave this one as it is, I think it "sounds" better this way, even if it'd be more consistent with your suggestion.

> The integration works by having k6 inject tracing metadata into the requests it sends to users’ backend services when they run a test. The tracing data is then correlated with k6 test run data, so users can understand how their services and operations behaved during the whole test run. The collected tracing data is aggregated to generate real-time metrics, such as frequency of calls, error rates, and percentile latencies, that help users narrow their search space and quickly spot anomalies.

Reviewer 2: ```suggestion
The integration works by having k6 inject tracing metadata into the requests it sends to users’ backend services when they run a test. The tracing data is then correlated with k6 test run data, so users can understand how their services and operations behaved during the whole test run. The collected tracing data is aggregated to generate real-time metrics—such as frequency of calls, error rates, and percentile latencies—that help users narrow their search space and quickly spot anomalies.
```

Reviewer 2: Wondering if we can change "users" to "you" in the paragraph above or are we talking about "your users"?

## Distributed Tracing in Grafana Cloud k6

You can now use the Grafana Cloud Traces integration with Grafana Cloud k6 to quickly debug failed performance tests and proactively improve application reliability.

Distributed tracing in Grafana Cloud k6 only requires two things:

- An application instrumented for tracing with Grafana Cloud Traces.
- Adding a few lines of code to your existing k6 scripts.

The integration works by having k6 inject tracing metadata into the requests it sends to users’ backend services when they run a test. The tracing data is then correlated with k6 test run data, so users can understand how their services and operations behaved during the whole test run. The collected tracing data is aggregated to generate real-time metrics, such as frequency of calls, error rates, and percentile latencies, that help users narrow their search space and quickly spot anomalies.

> To learn more, refer to the [Integration with Grafana Cloud Traces documentation](/docs/grafana-cloud/k6/analyze-results/integration-with-grafana-cloud-traces/) and [Distributed Tracing in Grafana Cloud k6 blog post](https://grafana.com/blog/2023/09/19/troubleshoot-failed-performance-tests-faster-with-distributed-tracing-in-grafana-cloud-k6/).

Reviewer: Are lines 47-58 the new one I assume?  That looks good!

> To learn more, refer to the [Integration with Grafana Cloud Traces documentation](/docs/grafana-cloud/k6/analyze-results/integration-with-grafana-cloud-traces/) and [Distributed Tracing in Grafana Cloud k6 blog post](https://grafana.com/blog/2023/09/19/troubleshoot-failed-performance-tests-faster-with-distributed-tracing-in-grafana-cloud-k6/).

Author: Thanks! Alright, I'll go ahead and update this PR with this version and get this merged. :shipit:
